### PR TITLE
feat: add -t and -i flags to docker run command so that docker contai…

### LINKED ISSUE #65: Ctrl+c doesn't exit devnet
## Context

I just a local devnet with the steps at https://github.com/rollkit/local-celestia-devnet#to-build-and-run-the-docker-image

## Problem

I'm attempting to stop the devnet with Ctrl + C but that's not stopping the devnet. Notice the ^C in the terminal output:

```
^C7:57PM INF Timed out dur=10993.078792 height=14 module=consensus round=0 step=1
7:57PM INF received proposal module=consensus proposal={"Type":32,"block_id":{"hash":"E3F0A48B96D9A48236EEECD0CA5D7B312F47620787844A13185BA1F51926E4A3","parts":{"hash":"653454D2A26B2FDF64CFEA23EC46B49F25F6F225545DB7ECA96B8756C402D587","total":1}},"height":14,"pol_round":-1,"round":0,"signature":"HvyIDuTbr/dftdMAaPkorWU/G+t+f4DVA1XErc1+0WuYXqCIrx1E6KaY7t/UWAFiMkMmW/SufCjdQxTc6rinAA==","timestamp":"2023-09-19T19:57:35.175358293Z"}
7:57PM INF received complete proposal block hash=E3F0A48B96D9A48236EEECD0CA5D7B312F47620787844A13185BA1F51926E4A3 height=14 module=consensus
```

To resolve I had to open Docker Dashboard and stop the container with 

Screenshot 2023-09-19 at 3 57 24 PM

## Question

Is this expected behavior? Is it possible for the devnet to stop if I Ctrl+C in terminal?

### PR DESCRIPTION
**Pull request by MSevey**
…ner can be stopped with crtl c

## Overview

Close #65 

REF: https://forums.docker.com/t/docker-run-cannot-be-killed-with-ctrl-c/13108/4

```
  -i, --interactive                    Keep STDIN open even if not attached
  -t, --tty                            Allocate a pseudo-TTY
```

### CODE REVIEW

**Review by jcstein — APPROVED** [review]
utACK

**Review by Manav-Aggarwal — DISMISSED** [review]

**Comment by nashqueue** [comment]
If you want to run a container in the background (no opened terminal required), use the parameter `-d`. You can still get the logs with `docker logs ...` and attach a terminal to a running container using `docker attach ...`.

_Originally posted by @smuu in https://github.com/rollkit/local-celestia-devnet/issues/65#issuecomment-1727875671_

Should we Add the flag -d ?

**Review by nashqueue — APPROVED** [review]
utAck

**Comment by smuu** [comment]
> If you want to run a container in the background (no opened terminal required), use the parameter `-d`. You can still get the logs with `docker logs ...` and attach a terminal to a running container using `docker attach ...`.
> 
> _Originally posted by @smuu in [#65 (comment)](https://github.com/rollkit/local-celestia-devnet/issues/65#issuecomment-1727875671)_
> 
> Should we Add the flag -d ?

I don't think `-d` should be added by default. This will confuse people running this. Maybe we can add a comment that if someone wants to run it in the background, they can start it with the argument added.

**Comment by MSevey** [comment]
> > If you want to run a container in the background (no opened terminal required), use the parameter `-d`. You can still get the logs with `docker logs ...` and attach a terminal to a running container using `docker attach ...`.
> > _Originally posted by @smuu in [#65 (comment)](https://github.com/rollkit/local-celestia-devnet/issues/65#issuecomment-1727875671)_
> > Should we Add the flag -d ?
> 
> I don't think `-d` should be added by default. This will confuse people running this. Maybe we can add a comment that if someone wants to run it in the background, they can start it with the argument added.

defer to @jcstein on this

**Comment by MSevey** [comment]
> I think I will follow the advice from @smuu and can add a note to the docs on using `-d` flag
> 
> @smuu can you confirm that the use of the `-i` and `-t` flags is something you think is worth adding?

If we no longer need to copy anything from the logs (which is a UX we should maintain imo) then we should add these flags so that users can stop the docker container with crtl c, since that is a standard practice. 

Otherwise users need to detach from the container and use `docker stop`. 

Either way we should include how to stop the container in the README. 

I'll update with my proposed solution.

File: README.md [hunk 1]
```diff
@@ -45,6 +45,41 @@ Test that the HTTP gateway server is up:
...
+    -p 26657:26657 -p 26658:26658 -p 26659:26659 -p 9090:9090 \
+    celestia-local-devnet
+```
+
+## Stopping the Container
+
+To stop the container when you are attached to the container, you can use
+`Ctrl+C` to stop the container.
+
+To stop the container when ou are detached from the container, you can use the
```
Comment (nashqueue) [inline README.md:64]: [nit]

```suggestion
To stop the container when you are detached from the container, you can use the
```

**Review by nashqueue — COMMENTED** [review]
LGTM

**Review by nashqueue — APPROVED** [review]
utAck

File: README.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ docker build . -t celestia-local-devnet
 To run the Docker container:
```
Comment (jcstein) [inline README.md:34]: on line 31, should it be `docker build . -t -i celestia-local-devnet` or without either flag `docker build . celestia-local-devnet`? @MSevey

**Review by jcstein — APPROVED** [review]
ACK

**Pull request merged**